---
title: "Class 15 Pertussis SO COOL!"
author: "WBray A69034838"
format: pdf
editor: visual
---
#This is an awesome topic!
#First create a data frame using the CDC data set linked.
#use datapasta to insert data frame from linked data
[here](https://www.cdc.gov/pertussis/php/surveillance/pertussis-cases-by-year.html?CDC_AAref_Val=https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html)
#So we want to web scrape this information from the website!
#install datapasta from CRAN, then use addins (or tools -> addins) to select this function!
#there's another package called rvst, but requires raw html and is therefore more effort to deal with
```{r}
cdc <- data.frame(
                          Year = c(1922L,
                                   1923L,1924L,1925L,1926L,1927L,1928L,
                                   1929L,1930L,1931L,1932L,1933L,1934L,1935L,
                                   1936L,1937L,1938L,1939L,1940L,1941L,
                                   1942L,1943L,1944L,1945L,1946L,1947L,1948L,
                                   1949L,1950L,1951L,1952L,1953L,1954L,
                                   1955L,1956L,1957L,1958L,1959L,1960L,
                                   1961L,1962L,1963L,1964L,1965L,1966L,1967L,
                                   1968L,1969L,1970L,1971L,1972L,1973L,
                                   1974L,1975L,1976L,1977L,1978L,1979L,1980L,
                                   1981L,1982L,1983L,1984L,1985L,1986L,
                                   1987L,1988L,1989L,1990L,1991L,1992L,1993L,
                                   1994L,1995L,1996L,1997L,1998L,1999L,
                                   2000L,2001L,2002L,2003L,2004L,2005L,
                                   2006L,2007L,2008L,2009L,2010L,2011L,2012L,
                                   2013L,2014L,2015L,2016L,2017L,2018L,
                                   2019L,2020L,2021L,2022L, 2024L),
  Cases = c(107473,
                                   164191,165418,152003,202210,181411,
                                   161799,197371,166914,172559,215343,179135,
                                   265269,180518,147237,214652,227319,103188,
                                   183866,222202,191383,191890,109873,
                                   133792,109860,156517,74715,69479,120718,
                                   68687,45030,37129,60886,62786,31732,28295,
                                   32148,40005,14809,11468,17749,17135,
                                   13005,6799,7717,9718,4810,3285,4249,
                                   3036,3287,1759,2402,1738,1010,2177,2063,
                                   1623,1730,1248,1895,2463,2276,3589,
                                   4195,2823,3450,4157,4570,2719,4083,6586,
                                   4617,5137,7796,6564,7405,7298,7867,
                                   7580,9771,11647,25827,25616,15632,10454,
                                   13278,16858,27550,18719,48277,28639,
                                   32971,20762,17972,18975,15609,18617,6124,
                                   2116,3044, 23544)
)

```
#installed Styler in otder to cleanup stuff! Didn't use it though
```{r}
library(ggplot2)
```

Question 1:
```{r}
baseplot <- ggplot(cdc) + aes(x= Year, y= Cases) + geom_point() + geom_line()

baseplot + geom_vline(xintercept = 1946, col = "red") + geom_vline(xintercept = 1996, col = "green") +
geom_vline(xintercept = 2020, col = "blue") +
geom_vline(xintercept = 2004, col = "yellow")
```
Question 2: The original vaccine was exceptionally effective; cases declined precipitously to almost nothing after introduction of the original vaccine. The aP vaccine may not be quite as effective, as we see some increase after its introduction, but this is also after a decade of anti-vaccine propaganda.
Question 3: The aP vaccine may not possess the same duration of protection as the original whole cell killed vaccine.

#look like the acellular vaccine has attenuated long term efficacy, would not have showed up in clinical trials since the phoneomena only appeared a decade after rollout. Why? This is where Barry and company come in...
#CMI-PB; can study individuals who had different types of vaccines to prime their immune response (boosting with aP vaccine)
#making data available to the public, has challenges for the scientific community...is HLA haplotype listed? (in fact they are doing whole genome sequencing, and have PBMC transcriptomics!)
#check understanidng the data section
#will need to check the API in order to pull down salient data.
# This project collects and makes available data abuot the immune responseto the Pertussis vaccine..
#Can be accessedvia API which returns JSON format (key:value pairs)
#therefore install JSOnlite package
```{r}
library(jsonlite)

subject <- read_json("http://cmi-pb.org/api/v5/subject", simplifyVector = TRUE)

```


```{r}
head(subject)
```
Question 4: how many subjects are in this dataset? - 172.
```{r}
nrow(subject)
```
```{r}
table(subject$biological_sex)
```
```{r}
table(subject$infancy_vac)
```
Question 5: 60 male, 112 female

#remember, table can do moultiple variables at one, but they are separated inside of the parentheticals by a comma, NOT nested
```{r}
table(subject$race, subject$biological_sex)
```
Question 6: - definitely does NOT reflect the US population overall; skewed toward UCSD students that needed the money and were willing to go into a hospital during the pandemic.
```{r}
table(subject$dataset)
```
#read in more data!
```{r}
specimen <- read_json("http://cmi-pb.org/api/v5/specimen", simplifyVector = TRUE)
ab_titer <- read_json("http://cmi-pb.org/api/v5/plasma_ab_titer", simplifyVector = TRUE)
PBMC <- read_json("http://cmi-pb.org/api/v5/pbmc_gene_expression?limit=25", simplifyVector = TRUE)



```
#let's check the head of these to see what commonalities we can find..
```{r}
head(specimen)
```
```{r}
head(ab_titer)
```
BARRY SKIPPED QUESTIONS 7/8

#practice some dplyr; let's combine these various table with the join command, we want antibody measurements combined with subject Id! Super cool, very important!
Question 9:
```{r}
library(dplyr)
meta <- inner_join(subject, specimen)
```
Question 10:
```{r}
abdata <- inner_join(meta, ab_titer)
nrow(abdata)
```
#want to make plots with the various variables
```{r}
head(abdata)
```
#we want to see how many different isotypes etc. in this file
Question 11:
```{r}
table(abdata$isotype)
```

```{r}
table(abdata$dataset)
```

Question 12: Values decline over time; not getting as many follow-up appointments as they would like!

```{r}
table(abdata$antigen)
```
#TT is tetanus toxoid, pertussis toxin is PT, FIM2/3 is filamentous hemaglutinin; don't want to see spikes in measles (ctrl)
#Let's begin our filtration wit IgG
```{r}
igg <- filter(abdata, isotype == "IgG")
head(igg)
```
Question 13:
```{r}
ggplot(igg) + aes(MFI_normalised, antigen) + geom_boxplot()
```

```{r}
ggplot(igg) + aes(MFI_normalised, antigen, col=infancy_vac) + geom_boxplot()
```
#we'd really like to see the time dependency in this case; specifically in relation to booster administration
```{r}
table(abdata$visit)
```
#try to utilize the facet wrap with the infancy data..
```{r}
ggplot(igg) + aes(MFI_normalised, antigen, col=infancy_vac) + geom_boxplot() + facet_wrap(~infancy_vac)
```
#Nice; ideally would be larger with transformed/scaled axes, but that's for another time.
Question 14: Responses decline with time, with the individuals responding positively to antigens in the booster!
```{r}
ggplot(igg) + aes(MFI_normalised, antigen, col=infancy_vac) + geom_boxplot() + facet_wrap(~visit)
```
Question 15:
```{r}
filter(igg, antigen=="OVA") %>%
  ggplot() +
  aes(MFI_normalised, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

```{r}
filter(igg, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI_normalised, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```

```{r}
filter(igg, antigen=="PT") %>%
  ggplot() +
  aes(MFI_normalised, col=infancy_vac) +
  geom_boxplot(show.legend = FALSE) +
  facet_wrap(vars(visit)) +
  theme_bw()
```
Q16: The PT response clearly increases post booster, but then declines once again to baseline. Ova remains rather elevated (continuous exposure, hence positive control antigen).

```{r}
anti <- filter(igg, antigen == "PT", dataset == "2021_dataset")
head(anti)
ggplot(anti) + aes(actual_day_relative_to_boost, MFI_normalised, col=infancy_vac, group = subject_id) + geom_point() + geom_line() + geom_smooth(se = FALSE, aes(group=infancy_vac), linewidth = 2) 
 
```
Question 17: Quite an interesting trend. There does appear to be a difference between acellular and whole Pertussis vaccines, as the latter seems to correlate with higher titers in response to the booster!
#submit as pdf
